app/src/main/java/com/piooda/recycrew/ui/main/home/HomeViewModel.kt

```diff
@@ -1,6 +1,69 @@
 package com.piooda.recycrew.ui.main.home
 
+import android.net.Uri
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.storage.FirebaseStorage
+
+// 데이터 클래스 정의
+data class ImageData(val title: String, val imageUrl: String, val num: Int)
 
 class HomeViewModel : ViewModel() {
-}
\ No newline at end of file
+
+    private val _imageList = MutableLiveData<List<ImageData>>()
+    val imageList: LiveData<List<ImageData>> = _imageList
+
```

Comment:
오 backing property 좋습니다!!

Reply:
추가로 이 LiveData의 경우 서버통신(fireabse)의 결과값을 가져오는 것이기 때문에
로딩, 실패, 성공 들의 상태 값들이 존재 합니다!

 Uistate를 제작하고, 이를 타입으로 받아 Uistate를 정의 해주는게 더욱 좋습니다!
```suggestion
    private val _imageList = MutableLiveData<Uistate<List<ImageData>>>()
    val imageList: LiveData<Uistate<List<ImageData>>> = _imageList
```

```kotlin
sealed interface UiState<out T> {

    data object Loading : UiState<Nothing>

    data class Success<T>(
        val data: T
    ) : UiState<T>

    data class Failure(
        val errorMessage: String
    ) : UiState<Nothing>
}
```
저는 주로 이런식으로  uistate를 제너릭타입으로 만들어서 사용하는 편입니다!

Reply:
또 LiveData와 StateFlow 의 차이점을 확인해보는 것도 좋을것 같아요~

LiveData의 가장 큰 단점은 
1. **메인스레드에서 작업이 일어난다는 점입니다**! 
즉 읽고 쓰기가 메인 스레드에서 일어난다는 점이죠.
이와 반대로 SateFlow는 비동기 스트림을 지원합니다. 
2. 안드로이드 라이브러리에 의존적
클린아키텍쳐의 경우 도메인 레이어는 순수 kotlin/java 코드로 이루어져야 하는데 LiveData는 이 점에서 자유롭지는 못합니다.
(물론 viewmodel은 ui 레이어이기 때문에 전혀 상관이 없습니다) 사실 StateFlow로 이전 하는 데에는 1번이유가 가장 큰거 같아요.

이와달리 SateFlow는 코틀린에 종속이이 때문에 안드로이드 생명주기를 모른다는 단점이 있습니다. 즉 뷰가 소멸이 되도 데이터 스트림의 업데이트가 생기면 계속 수집하여 메모리 leak이 일어 날 수 있습니다.
_-> 이 경우 어떻게 할까요?_  

> collect하는 부분에서 flowWithLifecycle 혹은 repeatOnLifecycle 사용하여 생명주기 동안 collect 하도록 하면 됩니다. 


여기서는 repository에서 flow를 emit하는 구조가 아니고 ui단에서 바로 업데이트가 일어나는 구조이기에 LiveData를 사용해도 무방하다 생각합니다!
다만 어떠한 차이점이 있는지 알아두시면 좋을것 같아요 :)

https://velog.io/@sana/Android-LiveData%EB%A5%BC-Flow%EB%A1%9C-%EB%8C%80%EC%B2%B4%ED%95%98%EA%B8%B0

https://oliveyoung.tech/blog/2022-12-14/Android-State-Flow/

Reply:
좋은 피드백 감사합니다. 실제로 코드를 확인하는 과정 중에 LiveData 사용 시 발생할 수 있는 문제점을 확인하였고, 이를 해결하기 위해 `StateFlow`로 코드를 수정할 예정입니다. 다시 한 번 감사합니다!